Dockerfile

```
@@ -1,22 +1,27 @@
+FROM node:16 AS prepare-node
+WORKDIR /app
+COPY ./package.json ./yarn.lock ./
+RUN yarn install --pure-lockfile
+
 # https://docs.docker.com/compose/rails/
 FROM ruby:2.7
-
-# https://github.com/nodesource/distributions/blob/master/README.md#installation-instructions
-RUN curl -sL https://deb.nodesource.com/setup_12.x | bash -
-RUN apt update -qq && apt install -y nodejs default-mysql-client && rm -rf /var/lib/apt/lists/*
-RUN npm install -g yarn
+RUN apt-get update -qq \
+    && apt-get install -y \
+    ca-certificates \
+    default-mysql-client \
+    nodejs \
+    npm \
+    && rm -rf /var/lib/apt/lists/*
 RUN gem install bundler -v 2.1.4
-
+RUN npm install -g yarn webpack-cli
```

```suggestion
RUN npm install -g yarn
```

I don't think it is necessary to install `webpack-cli` globally

---

As you say, webpack cli is not necessary.
I will remove it.